src/main/scala/loamstream/model/jobs/TerminationReason.scala

```diff
@@ -1,50 +1,64 @@
 package loamstream.model.jobs
 
+import scala.reflect.ClassTag
+import loamstream.util.Classes
+
 /**
  * @author clint
  * Apr 16, 2019
  * 
  * An ADT for the reasons why a job was killed by the underlying system (Uger/LSF/Google) 
  */
-sealed trait TerminationReason
+sealed abstract class TerminationReason(
+    val raw: Option[String], 
+    companion: TerminationReason.Companion[_]) {
+  
+  import TerminationReason._
+  
+  final def name: String = companion.Name
+  
+  final def isRunTime: Boolean = this.isInstanceOf[RunTime]
```

Comment:
Given that you have the companion object, and these are `final` classes, I wonder if it would just be easier to do:

```
final def isRunTime: Boolean = companion == TerminationReason.RunTime
```

Performance likely isn't a concern here, but if there's any weird hic-ups due to RTTI that can happen with `.isInstanceOf`, this would avoid them.

Reply:
I've gotten into the habit of implementing these sorts of flags with `isInstanceOf` since I don't have strong feelings about that method (at least in the implementation of trivial flags), and you've preferred it in past reviews.

That said, there _is_ a companion object here.  My gut feeling is that this is 6-and-half-dozen.